File: 07-memo-openid/src/gaebook/memo/New.java
package gaebook.memo;

import gaebook.util.PMF;

import java.io.IOException;
import java.util.Date;

import javax.jdo.PersistenceManager;
import javax.servlet.http.*;

import com.google.appengine.api.users.User;
import com.google.appengine.api.users.UserService;
import com.google.appengine.api.users.UserServiceFactory;

// 新しいメモをデータストアに保存するサーブレット 
public class New extends HttpServlet {
    public void doPost(HttpServletRequest req, HttpServletResponse resp)
            throws IOException {
        // User情報取り出し
        UserService userService = UserServiceFactory.getUserService();
        User user = userService.getCurrentUser();
        // リクエストからメモの中身を取り出し
        String content = req.getParameter("content");
        // Memo インスタンスを作成
        Memo memo = new Memo(user, content, new Date());

        PersistenceManager pm = PMF.get().getPersistenceManager();
        try {
            // データストアに保存
            pm.makePersistent(memo);
        } finally {
            pm.close();
        }
        // もとのページにリダイレクト
        resp.sendRedirect("/mainPage");
    }
}

File: 15-scala-memo/src/gaebook/memo/New.scala
package gaebook.memo

import java.io.IOException
import javax.servlet.http._
import gaebook.util._

import javax.jdo.PersistenceManager
import com.google.appengine.api.users.User
import com.google.appengine.api.users.UserService
import com.google.appengine.api.users.UserServiceFactory

class New extends HttpServlet {
  override def doPost(req: HttpServletRequest, resp: HttpServletResponse) {
    val user = UserServiceFactory.getUserService.getCurrentUser 

    val content = req.getParameter("content")
    val memo = new Memo(user, content, new java.util.Date())
    val pm = PMF.get.getPersistenceManager
    try {
      pm.makePersistent(memo) 
    } finally {
      pm.close
    }
    resp.sendRedirect("/mainPage")
  }
}
 
